### PR TITLE
[databases]: add support for Opensearch advanced configuration

### PR DESCRIPTION
**Pull request by loosla**
PR adds a support for Opensearch advanced configuration.

### CODE REVIEW

File: digitalocean/database/resource_database_opensearch_config_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/acceptance"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDigitalOceanDatabaseOpensearchConfig_Basic(t *testing.T) {
+	name := acceptance.RandomTestName()
+	dbConfig := fmt.Sprintf(testAccCheckDigitalOceanDatabaseClusterOpensearch, name, "3.7")
```
Comment (andrewsomething) [inline digitalocean/database/resource_database_opensearch_config_test.go:13]:
```suggestion
	dbConfig := fmt.Sprintf(testAccCheckDigitalOceanDatabaseClusterOpensearch, name, "2")
```

3.7 is not a supported Opensearch version:

```
$ doctl databases options versions
Engine        Versions
mysql         [8]
pg            [13,14,15,16]
redis         [7]
kafka         [3.7]
opensearch    [1,2]
mongodb       [5.0,6.0,7.0]
```

File: digitalocean/database/resource_database_cluster_test.go [hunk 2]
```diff
@@ -882,6 +882,17 @@ resource "digitalocean_database_cluster" "foobar" {
...
 
+const testAccCheckDigitalOceanDatabaseClusterOpensearch = `
+resource "digitalocean_database_cluster" "foobar" {
+  name       = "%s"
+  engine     = "opensearch"
+  version    = "%s"
+  size       = "db-s-2vcpu-2gb"
+  region     = "nyc1"
+  node_count = 3
+  tags       = ["production"]
```
Comment (andrewsomething) [inline digitalocean/database/resource_database_cluster_test.go:893]:
```suggestion
  size       = "db-s-1vcpu-2gb"
  region     = "nyc1"
  node_count = 1
  tags       = ["production"]
```

Not sure why, but looks like `db-s-2vcpu-2gb` isn't a supported size for Opensearch. While we're at it, we might as well use a single node configuration to keep the cost of running tests down.

```
$ doctl databases options slugs --engine opensearch
Slug              NodeNums
db-s-1vcpu-2gb    [1]
db-s-2vcpu-4gb    [1,3]
db-s-4vcpu-8gb    [1,3]
gd-2vcpu-8gb      [1,3,6,9,15]
gd-4vcpu-16gb     [1,3,6,9,15]
m3-2vcpu-16gb     [1,3,6,9,15]
m3-4vcpu-32gb     [1,3,6,9,15]
m3-8vcpu-64gb     [1,3,6,9,15]
```

File: digitalocean/database/resource_database_opensearch_config_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_history_max_age_hours", "10"),
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_history_max_docs", "1"),
+				),
+			},
+			{
+				Config: fmt.Sprintf(testAccCheckDigitalOceanDatabaseOpensearchConfigConfigBasic, dbConfig, false, 1, "9223372036854776000"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_enabled", "false"),
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_history_max_age_hours", "1"),
+					resource.TestCheckResourceAttr("digitalocean_database_opensearch_config.foobar", "ism_history_max_docs", "9223372036854776000"),
```
Comment (andrewsomething) [inline digitalocean/database/resource_database_opensearch_config_test.go:33]: Seems like this is an issue with the documentation, but if I try to use this value the API rejects it:

```
│ Error: Error updating Opensearch configuration: PATCH https://api.digitalocean.com/v2/databases/d62c44d8-772b-4876-9bd9-25a03115277b/config: 422 (request "16de8945-d643-463a-a848-321b5c83c177") invalid 'user_config': invalid input for opensearch: 9223372036854776000 is greater than the maximum of 9223372036854775807
```

**Review by andrewsomething — COMMENTED** [review]
Looking great, but I did notice a few issues with the acceptance test.

File: digitalocean/database/resource_database_opensearch_config_test.go (hunk 1 above)
Comment (loosla) [reply digitalocean/database/resource_database_opensearch_config_test.go:13]: Good catch! Thanks a lot, Andrew :+1:

File: digitalocean/database/resource_database_opensearch_config_test.go (hunk 3 above)
Comment (loosla) [reply digitalocean/database/resource_database_opensearch_config_test.go:33]: Decided to change the type for this field in godo from uint64 to int64.
The max value will be updated to 9223372036854775807 in our API doc.

Comment (loosla) [reply digitalocean/database/resource_database_opensearch_config_test.go]: @andrewsomething I updated the type here: [fa9a4ad](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1234/commits/fa9a4ad88bef8c687dd2ff442205421c110aa2dc) :pray: 
Thanks a lot for catching this :+1:

File: digitalocean/database/resource_database_opensearch_config.go [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+	if v, ok := d.GetOk("ism_history_enabled"); ok {
+		opts.IsmHistoryEnabled = godo.PtrTo(v.(bool))
+	}
+
+	if v, ok := d.GetOk("ism_history_max_age_hours"); ok {
+		opts.IsmHistoryMaxAgeHours = godo.PtrTo(v.(int))
+	}
+
+	if v, ok := d.GetOk("ism_history_max_docs"); ok {
+		opts.IsmHistoryMaxDocs = godo.PtrTo(v.(int64))
```
Comment (andrewsomething) [inline digitalocean/database/resource_database_opensearch_config.go:294]:
```suggestion
		opts.IsmHistoryMaxDocs = godo.PtrTo(int64(v.(int)))
```
I'm seeing a panic here:

```
Stack trace from the terraform-provider-digitalocean plugin:

panic: interface conversion: interface {} is int, not int64

goroutine 31 [running]:
github.com/digitalocean/terraform-provider-digitalocean/digitalocean/database.updateOpensearchConfig({0x12911d0, 0xc000159500}, 0xc000177e00, 0xc0003b4f08)
        /home/asb/projects/go/src/github.com/digitalocean/terraform-provider-digitalocean/digitalocean/database/resource_database_opensearch_config.go:294 +0x1aa5
```

**Review by andrewsomething — APPROVED** [review]
:+1: LGTM!

**Pull request merged**